### PR TITLE
don't delete media on ICDS

### PR DESCRIPTION
**Pull request by snopoke**
And change logging to log all in one email.

I want to just test this before merging

### CODE REVIEW

File: corehq/apps/hqmedia/models.py [hunk 1]
```diff
@@ -742,12 +746,19 @@ def remove_unused_mappings(self):
...
         for path in paths:
             if path not in permitted_paths:
-                map_changed = True
                 map_item = self.multimedia_map[path]
-                _log_media_deletion(self, map_item, path)
-                del self.multimedia_map[path]
+                deleted_media.append((path, map_item, None))
+                if not allow_deletion:
+                    map_changed = True
+                    del self.multimedia_map[path]
```
Comment (millerdev) [inline corehq/apps/hqmedia/models.py]: This is deleting only `if not allow_deletion`, which is confusing. Is that what you intended? Now this will only delete if the domain is 'icds-cas' or 'icds-test'. Seems like possibly not what you wanted.

Comment (snopoke) [reply corehq/apps/hqmedia/models.py]: yes, thanks. I swapped it around above but not here.

File: corehq/apps/hqmedia/models.py [hunk 2]
```diff
@@ -742,12 +746,19 @@ def remove_unused_mappings(self):
...
+        deleted_media = []
+        allow_deletion = self.domain not in {'icds-cas', 'icds-test'}
         for path in paths:
             if path not in permitted_paths:
-                map_changed = True
                 map_item = self.multimedia_map[path]
-                _log_media_deletion(self, map_item, path)
-                del self.multimedia_map[path]
+                deleted_media.append((path, map_item, None))
+                if not allow_deletion:
```
Comment (snopoke) [inline corehq/apps/hqmedia/models.py]:
```suggestion
                if allow_deletion:
```

File: corehq/apps/hqmedia/models.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/hqmedia/models.py]: updated

File: corehq/apps/hqmedia/models.py [hunk 3]
```diff
@@ -580,34 +584,35 @@ def _add_menu_media(item, **kwargs):
...
-                                if icon]
-                            )
-                # Print template
-                if display and details.display == 'long' and details.print_template:
-                    media.append(ApplicationMediaReference(
-                        details.print_template['path'],
-                        media_class=CommCareMultimedia,
-                        **media_kwargs)
-                    )
+            if module.module_type != 'report':
```
Comment (mkangia) [inline corehq/apps/hqmedia/models.py]: is that a safe change to skip if module_type is 'report'? Looks out of scope of the aim of this PR.

**Review by mkangia — APPROVED** [review]
Looks good pending one query.

Whats the next step after this? So, it wont delete mm but just send an email, how would that help find the underlying issue?

**Comment by snopoke** [comment]
I'm hoping so yes. In prior attempts to debug the deletion has already removed the media so it's been hard to re-produce. I'm hoping that we can now run the same code again to debug why the deletion is happening.

**Comment by mkangia** [comment]
@snopoke would be useful then to also get a list of permitted paths? exclusion from which causes the mm to be deleted?

File: corehq/apps/hqmedia/models.py (hunk 3 above)
Comment (snopoke) [reply corehq/apps/hqmedia/models.py]: yea, it is a bit out of scope. The alternative was to fix a bunch of tests that need reports to exists. I checked the code though and report modules only ever use custom_xml:
* https://github.com/dimagi/commcare-hq/blob/5cd850522dd807edbe811d366223c31f9fc05f09/corehq/apps/app_manager/suite_xml/features/mobile_ucr.py#L433
* https://github.com/dimagi/commcare-hq/blob/5cd850522dd807edbe811d366223c31f9fc05f09/corehq/apps/app_manager/suite_xml/features/mobile_ucr.py#L122
* https://github.com/dimagi/commcare-hq/blob/5cd850522dd807edbe811d366223c31f9fc05f09/corehq/apps/app_manager/suite_xml/features/mobile_ucr.py#L280-L289

I'll take another look at the tests.

**Comment by snopoke** [comment]
> @snopoke would be useful then to also get a list of permitted paths? exclusion from which causes the mm to be deleted?

It's a very long list for ICDS. I'm fine with this for now, if we aren't able to make progress we can add more detail.

File: corehq/apps/hqmedia/models.py (hunk 3 above)
Comment (snopoke) [reply corehq/apps/hqmedia/models.py]: reverted this commit and updated tests 87fb3defee6

**Review by millerdev — APPROVED** [review]
